cohort-engine-api-web/src/main/java/com/ibm/cohort/engine/api/service/CohortEngineRestStatusHandler.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * (C) Copyright IBM Corp. 2020, 2020
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package com.ibm.cohort.engine.api.service;
+
+import javax.ws.rs.Path;
+
+import com.ibm.watson.common.service.base.ServiceStatusHandler;
+import com.ibm.watson.service.base.model.ServiceStatus;
+import com.ibm.watson.service.base.model.ServiceStatus.ServiceState;
+
+import io.swagger.annotations.Api;
+import io.swagger.annotations.SwaggerDefinition;
+import io.swagger.annotations.Tag;
+
+@Path(CohortEngineRestConstants.SERVICE_MAJOR_VERSION+"/status")
+@Api(value = "Status")
+@SwaggerDefinition(tags={@Tag(name = "Status", description = "Get the status of this service")})
+public class CohortEngineRestStatusHandler extends ServiceStatusHandler {
+
+	/**
+	 * Example to show how to modify the service status text.
+	 * This method should include code that verifies the service
+	 * is running properly.
+	 */
+	@Override
+	public ServiceStatus adjustServiceStatus(ServiceStatus status) {
+
+		status.setServiceState(ServiceState.OK);
```

Comment:
Should the state of the service be set to the value passed in instead of OK always?

Also I'm intrigued by an endpoint that allows an external caller to update the state of the service - can you share a real-life use case for this?

Reply:
The actual liveness, readiness, and healthcheck endpoints are provided by the superclass ServiceStatusHandler, which is part of the rest service framework projects. The superclass endpoint methods call this method to allow the subclass to be in the implementing project and this is where the application specific logic for determining the state of the application would go. The endpoints themselves don't allow and external caller to pass in the service state. In the future we may want to add logic here to check if any configured FHIR servers, databases, or any other external services that are needed for successful operation of our service are up and can be reached.